File: benchmark/nodejs/substring_search.js
let text = "";
for (let i = 0; i < 1000; i = i + 1) {
    text = text + "abcdefghij";
}
const pattern = "defg";
const text_len = text.length;
const pat_len = pattern.length;
let count = 0;
const start = Date.now();
for (let t = 0; t <= text_len - pat_len; t = t + 1) {
    let match = true;
    for (let p = 0; p < pat_len; p = p + 1) {
        if (text[t + p] !== pattern[p]) {
            match = false;
            break;
        }
    }
    if (match) {
        count = count + 1;
    }
}
const end = Date.now();
console.log(`Count: ${count}`);
console.log(`Time: ${end - start}ms`);

File: benchmark/ruby/substring_search.rb
text = ""
for i in 1..1000
    text = text + "abcdefghij"
end
pattern = "defg"
text_len = text.length
pat_len = pattern.length
count = 0
start = Time.now.to_f * 1000
for t in 0..(text_len - pat_len)
    match = true
    for p in 0..(pat_len - 1)
        if text[t + p] != pattern[p]
            match = false
            break
        end
    end
    if match
        count = count + 1
    end
end
finish = Time.now.to_f * 1000
puts "Count: #{count}"
puts "Time: #{(finish - start).round}ms"
